torchgeo/datamodules/utils.py

```diff
@@ -52,3 +54,75 @@ def dataset_split(
             [train_length, val_length, test_length],
             generator=Generator().manual_seed(0),
         )
+
+
+def group_shuffle_split(
+    groups: Any,
```

Comment:
Not sure what the typing should be on this or the return values

Reply:
Seems like input can be any Sequence, and output is currently np.ndarray